packages/common/src/lib/notify/notify-builder.js

```diff
@@ -1,184 +1,209 @@
 const logger = require('../logger');
 const { createNotifyClient } = require('./notify-factory');
 
-function NotifyBuilder() {
-  function setPrivateNotifyClient(notifyClient) {
+module.exports = {
+  reset() {
+    this.notifyClient = undefined;
+    this.templateId = undefined;
+    this.templatePersonalisation = undefined;
+    this.destinationEmail = undefined;
+    this.emailReplyToId = undefined;
+    this.reference = undefined;
+
+    return this;
+  },
+
+  setPrivateNotifyClient(notifyClient) {
     logger.debug({ notifyClient }, 'Setting notify client');
     this.notifyClient = notifyClient;
-  }
+  },
 
   /**
    * Intentionally do not expose the notify client directly.
    *
    * @returns {*}
    */
-  function getNotifyClient() {
+  getNotifyClient() {
```

Comment:
comment on this function is no longer relevant 

reason for that approach was to avoid the consumer of the NotifyBuilder from accidentally attaching the file to the wrong instance. Probably not a worry in an internal library in hindsight. 

In this approach `setPrivateNotifyClient` logic could be merged here.